src/pveclib/vec_f128_ppc.h

```diff
@@ -1667,6 +1665,426 @@ static inline vec_xscvsqqp (vi128_t int128)
  * \subsubsection f128_softfloat_0_0_2_y Round to Quad-Precision Integer
  * TBD
  *
+ * \subsection f128_softfloat_0_0_3 Quad-Precision Arithmetic
+ *
+ * The POWER9 (PowerISA 3.0B) processor provides a full set of
+ * Quad-Precision arithmetic operations; add, divide, multiply,
+ * multiply-add/sub, and subtract.
+ * The compilers (that support IEEE128) provide normal C-language
+ * arithmetic operators for the __float128 (ISO _Float128) data type.
+ * The compiler will generate in-line quad-precision instructions for
+ * the (-mcpu=power9) target and calls to the (soft-float) runtime
+ * for earlier processor targets.
+ *
+ * POWER9 also provides a round-to-odd override for these operations.
+ * This helps software avoid <I>double rounding</I> errors
+ * when rounding to smaller precision. These are supported with
+ * compile built-ins (or in-line assembler).
+ *
+ * The compilers (that support IEEE128) provide a soft-float
+ * implementation for POWER8. However the runtime does not support
+ * round-to-odd as a rounding mode. Also the compiler built-ins for
+ * round-to-odd arithmetic are disabled for POWER8 and earlier.
+ *
+ * As a minimum the PVECLIB implementation should
+ * implement POWER8 equivalents to the compiler built-ins
+ * supported for ISA 3.0.
+ * This would include the explicit round-to-odd operations.
```

Comment:
Should these implementations be submitted to GCC?

Reply:
I think so, but that is a Bill question.

Reply:
The bigger question is the existing libgcc kf (not round to odd) implementations for POWER8. 

The performance difference is at least partially do to bad GCC code-gen for float128 <-> int128 transfers for the power8 target. This forces a load-hit-store reject.

I don't want to do the extra work for the other (4) rounding modes if this can be fixed in GCC.